### PR TITLE
Refactor phpunit dataprovider to be static

### PR DESCRIPTION
**Pull request by mamazu**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | -
| Related issues/PRs | #6988 
| License | MIT
| Documentation PR | -

#### What's in this PR?
Fixing more dataproviders.

### Todo
- [ ] src/Sulu/Component/Content/Tests/Unit/Document/Subscriber/PHPCR/SuluNodeTest.php

### CODE REVIEW

**Comment by mamazu** [comment]
@alexander-schranz Do you have an idea how to tackle the dataprovider in the `SuluNode`? Possible options:
- Reflect all objects that the delegator test needs.
- Only test a couple of the methods
- Splitting the datatprovider up in seperate test methods

**Comment by alexander-schranz** [comment]
@mamazu let me know when this is ready to review.

**Comment by alexander-schranz** [comment]
@mamazu I personally think the dataprovider in `SuluNode` does not make any sense and would replace it with own test methods. What do you think?

**Comment by mamazu** [comment]
Yes, this would make the test much easier as well.

File: src/Sulu/Component/Content/Tests/Unit/Document/Subscriber/PHPCR/SuluNodeTest.php [hunk 1]
```diff
@@ -24,74 +25,544 @@ class SuluNodeTest extends TestCase
...
+
+    public function testGetSharedSet(): void
+    {
+        $nodes = [];
+        $node = $this->prophesize(NodeInterface::class);
+        $node->getSharedSet()->willReturn($nodes);
+
+        $suluNode = new SuluNode($node->reveal());
+
+        $this->assertSame($nodes, iterator_to_array($suluNode->getSharedSet()));
```
Comment (alexander-schranz) [inline src/Sulu/Component/Content/Tests/Unit/Document/Subscriber/PHPCR/SuluNodeTest.php]: @mamazu if you want to convert a `iterable` (array / Traversable) to an array you can use `[...$suluNode->getSharedSet()]` this works with arrays and traversables.

**Review by alexander-schranz — APPROVED** [review]

**Comment by mamazu** [comment]
I think those should be all instances.

**Review by alexander-schranz — APPROVED** [review]

**Comment by alexander-schranz** [comment]
@mamazu Thank you. Already merged also to 2.6

**Pull request merged**